src/Microsoft.Data.SqlClient/netcore/src/Microsoft/Data/SqlClient/TdsParser.cs

```diff
@@ -2792,6 +2792,11 @@ private bool TryProcessDone(SqlCommand cmd, SqlDataReader reader, ref RunBehavio
                         cmd.InternalRecordsAffected = count;
                     }
                 }
+                // Skip the bogus DONE counts sent by the server
```

Comment:
I know you didn't put this comment here but do you know what it means? I understand the words but not how they describe the test that follows them.

Reply:
I've not stepped through/debugged, but I assume it's because the same code path is hit when the server sends the done signal after sending column metadata or other info to the driver. As for skipping it when curCmd != TdsEnums.SELECT, I'm unsure why only SELECT since there are other SQL statement tokens in the DONE packet (like INSERT/UPDATE/DELETE which I would think are relevant to OnStatementCompleted. But this change makes it the same as it was before the PR that introduced the issue. A good addition might be to include insert/update/delete statements in the new test to verify.

Reply:
The docs just say "Occurs when the execution of a Transact-SQL statement completes." which doesn't really match the behaviour, as you say it should probably be triggered by other statement types. It might be worth opening and separate issue to see what happens with other types (insert with output for example) and improve/extend or better document the behaviour.

The port of old code back in LGTM,

Reply:
Yeah, I kept it same as NetFx so we can make them even in future!

Reply:
Added more command scenarios to the test